src/math/polysat/solver.cpp

```diff
@@ -682,46 +716,105 @@ namespace polysat {
      * We add 'p == 0' as a lemma. The lemma depends on the dependencies used
      * to derive p, and the level of the lemma is the maximal level of the dependencies.
      */
-    void solver::learn_lemma(pvar v, scoped_clause&& lemma) {
+    void solver::learn_lemma(pvar v, clause_ref lemma) {
```

Comment:
clause_ref& ?

Reply:
I would prefer to keep passing by value. To avoid unnecessary reference counting, we can use `std::move` at the callsite. This also makes it clear at the callsite where the ownership will be "stolen".